.github/workflows/general.yml

```diff
@@ -112,4 +115,26 @@ jobs:
       - name: Build Static Web
         run: tool/gh_actions/devtool/build_web.sh
 
+      # 1. checkout to branch artifacts
+      # 2. Do not ignore build/ folder (just for artifacts branch)
+      # 3. Commit everythings including the build/ folder
+      # 4. Push the new branch artifacts
+      - name: Create artifact branch and commit
+        run: |
+          git config --global user.name "${GITHUB_ACTOR}"
+          git config --global user.email "${GITHUB_ACTOR}@users.noreply.github.com"
+          git fetch origin
+          git checkout -b artifacts
+          git pull origin main
+          sed -i '/build\//d' .gitignore  # Remove the line that ignores the build directory
+          git add .
+          git commit -m "Add build artifacts and modify .gitignore"
+
+      - name: Push changes
```

Comment:
the push should only happen if there's a push to main, not for pull requests!  otherwise every PR will push immediately to this branch